common/nl.c

```diff
@@ -258,20 +239,24 @@ nl_sock_new(int protocol)
 	if (ret->fd < 0)
 		goto err;
 
-	if (protocol == NETLINK_KOBJECT_UEVENT) {
-		if (nl_sock_conf_uevent_sock(ret))
-			goto err;
-	} else if (protocol == NETLINK_ROUTE) {
-		if (nl_sock_conf_route_sock(ret))
-			goto err;
-	} else if (protocol == NETLINK_XFRM) {
-		if (nl_sock_conf_xfrm_sock(ret))
-			goto err;
-	} else { // use rtnetlink config as default
-		if (nl_sock_conf_route_sock(ret))
-			goto err;
+	switch (protocol) {
+		case NETLINK_KOBJECT_UEVENT:
+			if (nl_sock_conf_uevent_sock(ret)) {
```

Comment:
why not just move the groups parameter through to the particular nl_sock_conf_*_sock() methods
and do set/append the ml_groups there?

Reply:
Ok sounds reasonable to me, would you then please describe that in the commit message?
So just coping your comment above into the commit message would do it.
Word-wrap 75 chars at most please.

Thanks,
Michael

Reply:
Thank you for the review. Adopted your suggestion to the commit message.

Reply:
ok one more thing. code-style check faild for your contribution. Just run the formatting script in the repo:
`./scripts/format-code.sh`

Reply:
We need and will add some guidelines to the README.md soon. ;)

Reply:
done.